### PR TITLE
Thread.sleep() method is replaced with Awaitility

### PR DESCRIPTION
**Pull request by shreyanshR7**
@nastra

### CODE REVIEW

File: flink/v1.15/flink/src/test/java/org/apache/iceberg/flink/source/TestStreamingMonitorFunction.java [hunk 1]
```diff
@@ -111,9 +112,10 @@ public void testConsumeWithoutStartSnapshotId() throws Exception {
       TestSourceContext sourceContext = new TestSourceContext(latch);
       runSourceFunctionInTask(sourceContext, function);
 
-      Assert.assertTrue(
-          "Should have expected elements.", latch.await(WAIT_TIME_MILLIS, TimeUnit.MILLISECONDS));
-      Thread.sleep(1000L);
+      Awaitility.await()
+        .atMost(WAIT_TIME_MILLIS, TimeUnit.MILLISECONDS)
+        .pollInterval(100, TimeUnit.MILLISECONDS)
+        .until(() -> latch.getCount() == 0);
```
Comment (nk1506) [inline flink/v1.15/flink/src/test/java/org/apache/iceberg/flink/source/TestStreamingMonitorFunction.java:118]: I don't think this is a good idea.  There are high changes to become flaky tests. 
At bad day, your latch count became 0 and threads are still doing remaining processing. but your worker thread will start compiling next lines. 
I think let `latch.await` be there . Just replace Thread.sleep with `Awaitility` .

File: flink/v1.15/flink/src/test/java/org/apache/iceberg/flink/source/TestStreamingMonitorFunction.java [hunk 2]
```diff
@@ -148,9 +150,10 @@ public void testConsumeFromStartSnapshotId() throws Exception {
...
```
Comment (nk1506) [inline flink/v1.15/flink/src/test/java/org/apache/iceberg/flink/source/TestStreamingMonitorFunction.java:156]: same as above.

File: flink/v1.16/flink/src/test/java/org/apache/iceberg/flink/source/TestIcebergSourceContinuous.java [hunk 3]
```diff
@@ -325,7 +326,7 @@ public void testSpecificSnapshotTimestamp() throws Exception {
     long snapshot0Timestamp = tableResource.table().currentSnapshot().timestampMillis();
 
     // sleep for 2 ms to make sure snapshot1 has a higher timestamp value
-    Thread.sleep(2);
+    Awaitility.await().pollDelay(2, TimeUnit.MILLISECONDS).until(() -> System.currentTimeMillis()-snapshot0Timestamp>2);
```
Comment (nk1506) [inline flink/v1.16/flink/src/test/java/org/apache/iceberg/flink/source/TestIcebergSourceContinuous.java:329]: IMHO, let's not use `Awaitility` for just system clock delay. We can enhance [TestHelpers](https://github.com/apache/iceberg/blob/dd02085b12e9b509efef2d9a8b9fe730c0c60b03/api/src/test/java/org/apache/iceberg/TestHelpers.java#L57)  like
`public static long waitUntil1ms() {
long previous = System.currentTimeMillis();
    long current = System.currentTimeMillis();
    while (current <= previous) {
      current = System.currentTimeMillis();
    }
    return current;
  }`
for any other `waitUntilAfter` we can use  `Awaitility`

**Review by nk1506 — CHANGES_REQUESTED** [review]

**Comment by shreyanshR7** [comment]
I've made the required changes :) @nk1506 @nastra

File: flink/v1.15/flink/src/test/java/org/apache/iceberg/flink/source/TestStreamingMonitorFunction.java [hunk 4]
```diff
@@ -113,7 +114,7 @@ public void testConsumeWithoutStartSnapshotId() throws Exception {
 
       Assert.assertTrue(
           "Should have expected elements.", latch.await(WAIT_TIME_MILLIS, TimeUnit.MILLISECONDS));
-      Thread.sleep(1000L);
+      Awaitility.await().atLeast(1, TimeUnit.SECONDS).until(() -> true);
```
Comment (nastra) [inline flink/v1.15/flink/src/test/java/org/apache/iceberg/flink/source/TestStreamingMonitorFunction.java:117]: this goes back to my earlier comment [here](https://github.com/apache/iceberg/pull/8715#issuecomment-1748190639). We don't want to blindly just change Thread.sleep usage to Awaitility

Comment (shreyanshR7) [reply flink/v1.15/flink/src/test/java/org/apache/iceberg/flink/source/TestStreamingMonitorFunction.java:117]: will it be ok if i replace this with simply Awaitility.await()
        .atMost(WAIT_TIME_MILLIS, TimeUnit.MILLISECONDS)
        .pollInterval(100, TimeUnit.MILLISECONDS)
        .until(() -> latch.getCount() == 0);

**Comment by nastra** [comment]
Closing this one as it has been superseded by https://github.com/apache/iceberg/pull/8804

**Pull request closed without merging**